### PR TITLE
Updated discovery_rule entity with org & location

### PR DESCRIPTION
**Pull request by sghai**

### CODE REVIEW

**Comment by sghai** [comment]
test results:
```console
In [10]: rule = DiscoveryRule(server_config, name='nailrule12', search_='cpu_count=1', hostgroup='hh_rhel7', location=[loc], organization=[org]).create()

In [11]: rule.name
Out[11]: u'nailrule12'
```

File: nailgun/entities.py [hunk 1]
```diff
@@ -755,6 +757,18 @@ def create_payload(self):
...
+
+        For more information, see `Bugzilla #1381129
+        <https://bugzilla.redhat.com/show_bug.cgi?id=1381129>`_.
+
+        """
+        return type(self)(
+            self._server_config,
+            id=self.create_json(create_missing)['id'],
+        ).read()
+
```
Comment (sghai) [inline nailgun/entities.py]: I was getting attribute error with POST & GET so creating entity with `create_json(create_missing=True)`

File: nailgun/entities.py [hunk 2]
```diff
@@ -779,6 +793,16 @@ def read(self, entity=None, attrs=None, ignore=None):
...
+    def update(self, fields=None):
+        """Fetch a complete set of attributes for this entity.
+
+        For more information, see `Bugzilla #1381129
+        <https://bugzilla.redhat.com/show_bug.cgi?id=1381129>`_.
+
+        """
+        self.update_json(fields)
+        return self.read()
+
```
Comment (sghai) [inline nailgun/entities.py]: Similar issue with PUT request

**Comment by elyezer** [comment]
Changes look good to me

**Comment by sghai** [comment]
@oshtaier @abalakh : Hey, please review and help to merge the changes. I thought it already merged. The changes required for tests in SatelliteQE/robottelo/pull/3881

**Comment by oshtaier** [comment]
ACK
@abalakh please, review quickly

**Comment by svtkachenko** [comment]
ACK but @sghai can you please add this methods to tests? `CreateTestCase.test_generic` and `UpdateTestCase.test_generic` should be enough.

**Comment by sghai** [comment]
@svtkachenko : thank you. Your suggestion fixed the coverage..

**Pull request merged**